### PR TITLE
Issue 6971: Implement conditional NoOp chunk storage

### LINKED ISSUE #6971: Implement conditional NoOp chunk storage
**Is your feature request related to a problem? Please describe.**
For performance testing we need an implementation that ignores all user provided data. In such cases it always returns zeroed out byte arrays when read. However does preservs the data under _system folder.

**Describe the solution you'd like**
We need to create a storage wrapper which takes the user provided ChunkStorage and also holds the NoOpChunkStorage. Depending upon whether path of chunk either forward to user chunk storage or no op chunkstorage.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Amit-Singh40**
**Change log description**  
Implements conditional NoOp chunk storage which takes ChunkStorage and it also holds NoOpChunkStorage. 

**Purpose of the change**  
Fixes #6971 

**What the code does**  
Depending upon whether path of chunk either forward to user chunk storage or no op chunkstorage.

**How to verify it**  
All test cases should pass and standalone deployment should work.

### CODE REVIEW

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    @Override
+    public CompletableFuture<Integer> write(ChunkHandle handle, long offset, int length, InputStream data) {
+        return isMetadataCall(handle) ? inner.write(handle, offset, length, data) : noOpChunkStorage.write(handle, offset, length, data);
+    }
+
+    @Override
+    public CompletableFuture<Integer> concat(ConcatArgument[] chunks) {
+        return inner.concat(chunks);
```
Comment (sachin-j-joshi) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java:119]: This is missing isMetadataCall check
should be.
`return isMetadataCall(handle) ? inner.concat(chunks) : noOpChunkStorage.concat(chunks);`

Also not sure why this method is not covered

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    @Override
+    public void report() {
+        inner.report();
+    }
+
+    @Override
+    public void close() throws Exception {
+        this.noOpChunkStorage.close();
```
Comment (sachin-j-joshi) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java:144]: what happens if `this.noOpChunkStorage.close();` throws

```
may be something like this ?
try {
this.noOpChunkStorage.close();
} finally {
this.inner.close();
}
```

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpStorageFactory.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Creates an instance of {@link  ConditionalNoOpStorageFactory} and delegates calls to inner {@link SimpleStorageFactory}.
+ */
+@RequiredArgsConstructor
+public class ConditionalNoOpStorageFactory implements SimpleStorageFactory {
+
+    @Getter
+    final protected ScheduledExecutorService executor;
```
Comment (sachin-j-joshi) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpStorageFactory.java:37]: Does this have to be protected?

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
Some minor changes.
Otherwise already looking good.

Thanks

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java (hunk 2 above)
Comment (Amit-Singh40) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java]: Done

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpStorageFactory.java (hunk 3 above)
Comment (Amit-Singh40) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpStorageFactory.java:37]: Removed this

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java (hunk 1 above)
Comment (Amit-Singh40) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java]: Done

**Review by sachin-j-joshi — APPROVED** [review]
LGTM.

**Pull request merged**